deploy/webhook.yaml

```diff
@@ -0,0 +1,265 @@
+              name: webhook
+          resources:
+            limits:
+              memory: 50Mi
+              cpu: 300m
+            requests:
+              memory: 50Mi
+              cpu: 300m
+          volumeMounts:
+            - name: webhook-certs
+              mountPath: /etc/certs
+              readOnly: true
+            - name: logs
+              mountPath: /tmp
+          securityContext:
+            readOnlyRootFilesystem: true
+      volumes:
+        - name: webhook-certs
+          secret:
+            secretName: service-apis-admission
+        - name: logs
+          emptyDir: {}
+---
+# TODO(cmluciano): Use helm-based configuration and move cert related
+# jobs/RBAC out of the main chart. This will make it easier to just
+# deploy the necessary webhook bits if an admin uses something like
+# cert-manager.
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: service-apis-admission
+  annotations:
+  labels:
+    name: service-apis-webhook
+    version: 0.0.1
+  namespace: service-apis
+spec:
+  template:
+    metadata:
+      name: service-apis-admission-create
+      labels:
+        name: service-apis-webhook
+        version: 0.0.1
+    spec:
+      containers:
+        - name: create
+          image: docker.io/jettech/kube-webhook-certgen:v1.5.0
+          imagePullPolicy: IfNotPresent
+          args:
+            - create
+            - --host=service-apis-admission,service-apis-admission.service-apis.svc
+            - --namespace=service-apis
+            - --secret-name=service-apis-admission
+          env:
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+      restartPolicy: OnFailure
+      serviceAccountName: service-apis-admission
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 2000
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: service-apis-admission-patch
+  labels:
+    name: service-apis-webhook
+    version: 0.0.1
+  namespace: service-apis
+spec:
+  template:
+    metadata:
+      name: service-apis-admission-patch
+      labels:
+        name: service-apis-webhook
+        version: 0.0.1
+    spec:
+      containers:
+        - name: patch
+          image: docker.io/jettech/kube-webhook-certgen:v1.5.0
+          imagePullPolicy: IfNotPresent
+          args:
+            - patch
+            - --webhook-name=service-apis-admission
+            - --namespace=service-apis
+            - --patch-mutating=false
+            - --patch-validating=true
+            - --secret-name=service-apis-admission
+            - --patch-failure-policy=Fail
+          env:
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+      restartPolicy: OnFailure
+      serviceAccountName: service-apis-admission
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 2000
```

Comment:
This is some good magic to get this setup.
We might need to think through about automatically setting up the webhook, as some users might want more control over it. I think it is fine to start with what we have in this patch right now.

Reply:
Yes that sounds good to me too and aligns with Rob's comment as well. I plan to move all of this to a helm based deployment once I get all pieces of the webhook merged.